docs/adoc/config-site.adoc

```diff
@@ -0,0 +1,278 @@
+====
+Whenever you change any aspect of the site configuration file, you must restart the `*docserv*` process.
+This is unlike all other {ds2} configuration which will be re-evaluated during each build instruction.
+====
+
+
+[#config-site-server-section]
+## The `[server]` section
+
+The server section defines global attributes for {ds2}.
+It accepts the following attributes:
+
+[source,ini]
+----
+[server]
+host = localhost
+port = 8080
+loglevel = 2
+repo_dir = /home/docserv/docserv-repos/
+temp_repo_dir = /home/docserv/docserv-temp-repos/
+max_threads = 8
+valid_languages = en-us de-de fr-fr
+----
+
+`[server]`::
+  The section identifier.
+
+`host` and `port`::
+  Specify using these attributes where to run the API endpoint of {ds2}.
+  The API endpoint is used to trigger builds and check build status.
+
+`loglevel`::
+  Specifies which types of messages to log via `stdout` (when running interactively) or on the system log (when running as a daemon).
+  Allows the values `0` (warnings only), `1` (warnings and informational messages), or `2` (all messages including debugging messages).
+
+`repo_dir`::
+  Specifies the directory that is used to permanently cache remote Git repositories.
+  Git repositories in this directory persist across builds and instances of the {ds2} process.
+
+`temp_repo_dir`::
+  Specifies the directory that is used to temporarily cache specific Git branches.
+  When receiving a build instruction, {ds2} will clone an individual branch of a Git repository from `repo_dir`.
+  It will then build using the data from `temp_repo_dir`.
+  This prevents branch checkout collisions during builds and other problems.
+  After the build instruction has run through,
+
+`max_threads`::
+  Specifies the maximum number of threads to use for {ds2}.
+  You can specify either a number, such as `4` or use `max` to use as many threads as there are logical CPU cores.
++
+In any case, the upper limit to the number of threads is the number of logical CPU cores.
+
+`valid_languages`::
+  Specifies the overall list of language codes recognized as valid by this instance of {ds2}.
+  Language codes must follow the schema `la-ng` (where `la` is the language and `ng` is the regional variety).
+  Languages codes must be specified separated by spaces.
++
+[NOTE]
+.Independent sets of language code definitions
+====
+{ds2} uses two independent sets of language code definitions:
+
+* A set of language codes valid for documents and navigational pages (this attribute, global)
+* A set of language codes that navigational pages are created for (`languages` attribute, target-specific)
+====
+
+
+[#config-site-target-section]
+## Target sections
+
+{ds2} supports reuse of the same `*docserv*` process to synchronize to multiple target servers.
+Each of those target servers is defined in its own target section.
+All or a subset of target servers can share certain parts of their configuration by sharing the specific configuration directories, if desired.
```

Comment:
I would also mention that some keys allows *relative paths*. In the list below I would mention which keys allows this.

Reply:
Done.